src/content/components/FlameChartCanvas.js

```diff
@@ -1,19 +1,45 @@
-import React, { Component, PropTypes } from 'react';
+// @flow
+import React, { Component } from 'react';
 import shallowCompare from 'react-addons-shallow-compare';
 import { timeCode } from '../../common/time-code';
 import TextMeasurement from '../../common/text-measurement';
 
+import type { Thread } from '../../common/types/profile';
+import type { Milliseconds, CssPixels, UnitIntervalOfProfileRange, DevicePixels } from '../../common/types/units';
+import type { StackTimingByDepth } from '../stack-timing';
+
+type Props = {
+  thread: Thread,
+  interval: Milliseconds,
+  rangeStart: Milliseconds,
+  rangeEnd: Milliseconds,
+  className: string,
+  containerWidth: CssPixels,
+  containerHeight: CssPixels,
+  viewportLeft: UnitIntervalOfProfileRange,
+  viewportRight: UnitIntervalOfProfileRange,
+  viewportTop: CssPixels,
+  viewportBottom: CssPixels,
+  stackTimingByDepth: StackTimingByDepth,
+  rowHeight: CssPixels,
+};
+
 const ROW_HEIGHT = 16;
 const TEXT_OFFSET_START = 3;
 const TEXT_OFFSET_TOP = 11;
 
 class FlameChartCanvas extends Component {
 
-  constructor(props) {
+  _requestedAnimationFrame: boolean
+  _devicePixelRatio: number
+  _textMeasurement: null|TextMeasurement
+  _ctx: null|CanvasRenderingContext2D
```

Comment:
There is no `props: Props` field here, but there is one in FlameChartViewPort. What's the right way to do this? Does flow automatically generate propTypes for us? Will those be checked at runtime?